src/js/jsx/search/SearchBar.jsx

```diff
@@ -250,41 +303,232 @@ define(function (require, exports, module) {
+                    };
+                }),
+
+                // To search for all layers, documents, etc
+                headerTitle = CATEGORIES[header.toUpperCase()];
+            headerTitle = headerTitle.charAt(0).toUpperCase() + headerTitle.slice(1);
+            var headerFilter = {
+                id: "filter_" + header,
+                title: headerTitle,
+                category: [CATEGORIES[header.toUpperCase()]],
+                type: "item"
+            };
+            
+            filters = filters.unshift(headerFilter);
 
-            return recentDocMap;
+            return filters;
         },
 
         /**
          * Make list of items and headers to be used as dropdown options
          * @return {Array.<object>}
          */
-        _getSelectOptions: function () {
-            var layerOptions = this._getLayerOptions(),
-                currentDocOptions = this._getCurrDocOptions(),
-                recentDocOptions = this._getRecentDocOptions();
+        _getAllSelectOptions: function () {
+            var filterOptions = this._getFilterOptions("layer").concat(this._getFilterOptions("document")),
+                layerOptions = this._getLayerOptions(),
+                docOptions = this._getCurrDocOptions().concat(this._getRecentDocOptions());
            
-            return layerOptions.concat(currentDocOptions).concat(recentDocOptions);
+            return filterOptions.concat(layerOptions).concat(docOptions);
+        },
+
+        /**
+         * Find the icons corresponding with the filter
+         *
+         * @private
+         * @param {Array.<string>} filter
+         * @return {Array.<string>}
+         */
+        _getFilterIcons: function (filter) {
+            // currently only have icons for layers
+            if (filter.length > 1 && filter.join(" ").indexOf("layer") > -1) {
+                return svgUtil.getSVGFromLayerType(filter);
+            } else {
+                return ["tool-rectangle"]; // standin for non-layers
+            }
+        },
+
+        /**
+         * Find options to show in the Datalist drop down
+         *
+         * @param {Array.<object>} options Full list of potential options
+         * @param {string} searchTerm Term to filter by
+         * @return {Array.<object>}
+         */
+        _filterSearch: function (options, searchTerm) {
+            // Keep track of how many options shown so far in a given category
+            var count = 0;
+
+            return options && options.filter(function (option) {
+                if (option.hidden) {
+                    return false;
+                }
+
+                // Always add headers to list of searchable options
+                // The check to not render if there are no options below it is in Select.jsx
+                if (option.type === "header") {
+                    count = 0;
+                    return true;
+                }
+
+                if (count === MAX_OPTIONS) {
```

Comment:
Since you're limiting to a relatively small `MAX_OPTIONS`, it might be worthwhile to short-circuit this search once the list is full. You could use `.some` instead of `.filter` for that.